aws/resource_aws_backup_selection_test.go

```diff
@@ -0,0 +1,189 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/backup"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAwsBackupSelection_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsBackupSelectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBackupSelectionConfigBasic(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsBackupSelectionExists("aws_backup_selection.test"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAwsBackupSelection_withTags(t *testing.T) {
+	rInt := acctest.RandInt()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsBackupSelectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBackupSelectionConfigWithTags(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsBackupSelectionExists("aws_backup_selection.test"),
+					resource.TestCheckResourceAttr("aws_backup_selection.test", "tag.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAwsBackupSelection_withResources(t *testing.T) {
+	rInt := acctest.RandInt()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsBackupSelectionDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBackupSelectionConfigWithResources(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsBackupSelectionExists("aws_backup_selection.test"),
+					resource.TestCheckResourceAttr("aws_backup_selection.test", "resources.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsBackupSelectionDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).backupconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_backup_selection" {
+			continue
+		}
+
+		input := &backup.GetBackupSelectionInput{
+			BackupPlanId: aws.String(rs.Primary.Attributes["plan_id"]),
+			SelectionId:  aws.String(rs.Primary.ID),
+		}
+
+		resp, err := conn.GetBackupSelection(input)
+
+		if err == nil {
+			if *resp.SelectionId == rs.Primary.ID {
+				return fmt.Errorf("Selection '%s' was not deleted properly", rs.Primary.ID)
+			}
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAwsBackupSelectionExists(name string) resource.TestCheckFunc {
```

Comment:
The acceptance testing `Exists` functions should be performing an API call to verify the resource creation did occur and preferably return the API response object for potential usage in other checking functions. In this case it should be calling `GetBackupSelection` similar to the `Destroy` function above.

```go
func testAccCheckAwsBackupSelectionExists(name string, selection *backup.GetBackupSelectionOutput) resource.TestCheckFunc {
	return func(s *terraform.State) error {
		_, ok := s.RootModule().Resources[name]
		if !ok {
			return fmt.Errorf("not found: %s, %v", name, s.RootModule().Resources)
		}

		conn := testAccProvider.Meta().(*AWSClient).backupconn

		input := &backup.GetBackupSelectionInput{
			BackupPlanId: aws.String(rs.Primary.Attributes["plan_id"]),
			SelectionId:  aws.String(rs.Primary.ID),
		}

		output, err := conn.DescribeBackupSelection(input)

		if err != nil {
			return err
		}

		*selection = *output

		return nil
	}
}
```